frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.js

```diff
@@ -0,0 +1,185 @@
+// eslint-disable-next-line no-restricted-imports -- deprecated usage
+import moment from "moment-timezone";
+import { t } from "ttag";
+import * as Lib from "metabase-lib";
+import { formatValue } from "metabase/lib/formatting/value";
+import { formatDateTimeRangeWithUnit } from "metabase/lib/formatting/date";
+import { color } from "metabase/lib/colors";
+import { formatChange } from "metabase/visualizations/visualizations/SmartScalar/utils";
+import { isDate } from "metabase-lib/types/utils/isa";
+
+const FALLBACK_DATE_UNIT = "day";
+
+// compute the percent change between two values (a → b)
+export function computeChange(a, b) {
```

Comment:
ported from clojure: https://github.com/metabase/metabase/blob/d2ecd17/src/metabase/sync/analyze/fingerprint/insights.clj#L32-L42

For readability, I split up the cases into comments, which are validated in [unit tests below](https://github.com/metabase/metabase/pull/35184/files#diff-5319611e9f4c7a99acd8792fbeb80a7961e52432a545beaad54ab259db37c4b2R8-R59).